### DISCUSSION TITLE
mysql - sql_require_primary_key

### DISCUSSION THREAD

**Discussion by azGreg** [OrchardCMS/OrchardCore#11271] [Q&A]
I'm trying to create a system connecting to a MySQL database using OrchardCore 1.2.2 NuGet packages.

The database is hosted on DigitalOcean. I end up with an empty database (no tables) and sql_require_primary_key errors.

It appears that the host requires primary keys for management (https://docs.digitalocean.com/products/databases/mysql/how-to/create-primary-keys/)

Is this supported in a newer preview version?

**Comment by hishamco** [comment]
You can set primary keys through`SchemaBuilder`, it's migrations responsibility to set them for the tables according the requirements

**Reply by azGreg** [reply]
When I install to MySql I get no tables. It is difficult to manage them.
I am unfamiliar with SchemaBuilder.

**Reply by azGreg** [reply]
I switched away from DigitalOcean. No problem with Linode.

**Reply by hishamco** [reply]
I don't think there's no PKs in OC
